drivers/SmartThings/zigbee-switch/src/multi-switch-no-master/init.lua

```diff
@@ -11,61 +11,115 @@
+  { mfr = "eWeLink", model = "ZB-SW02", children = 2 },
+  { mfr = "eWeLink", model = "ZB-SW03", children = 3 },
+  { mfr = "eWeLink", model = "ZB-SW04", children = 4 },
+  { model = "E220-KR2N0Z0-HA", children = 1 },
+  { model = "E220-KR3N0Z0-HA", children = 2 },
+  { model = "E220-KR4N0Z0-HA", children = 3 },
+  { model = "E220-KR5N0Z0-HA", children = 4 },
+  { model = "E220-KR6N0Z0-HA", children = 5 }
 }
 
 local function is_multi_switch_no_master(opts, driver, device)
   for _, fingerprint in ipairs(MULTI_SWITCH_NO_MASTER_FINGERPRINTS) do
-      if device:get_manufacturer() == nil and device:get_model() == fingerprint.model then
-        return true
-      elseif device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
-        return true
-      end
+    if device:get_manufacturer() == nil and device:get_model() == fingerprint.model then
+      return true
+    elseif device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+      return true
+    end
   end
-
   return false
 end
 
-local function component_to_endpoint(device, component_id)
-  local ep_num = component_id:match("switch(%d)")
-  return ep_num and tonumber(ep_num) + 1 or 1
+local function get_children_amount(device)
+  for _, fingerprint in ipairs(MULTI_SWITCH_NO_MASTER_FINGERPRINTS) do
+    if device:get_model() == fingerprint.model then
+      return fingerprint.children
+    end
+  end
+end
+
+local function do_refresh(self, device)
+  device:send(OnOff.attributes.OnOff:read(device))
+end
+
+local function device_added(driver, device, event)
+  if device.network_type == st_device.NETWORK_TYPE_ZIGBEE then
+    local children_amount = get_children_amount(device)
+    for i = 2, children_amount+1, 1 do
+      local device_name_without_number = string.sub(device.label, 0,-2)
+      local name = string.format("%s%d", device_name_without_number, i)
+      local metadata = {
+        type = "EDGE_CHILD",
+        label = name,
+        profile = "multi-switch-no-master",
+        parent_device_id = device.id,
+        parent_assigned_child_key = string.format("%02X", i),
+        vendor_provided_label = name,
+      }
+      driver:try_create_device(metadata)
+    end
+  end
+  do_refresh(driver, device)
 end
 
-local function endpoint_to_component(device, ep)
-  local switch_comp = string.format("switch%d", ep - 1)
-  if device.profile.components[switch_comp] ~= nil then
-    return switch_comp
+local function find_child(parent, ep_id)
+  if ep_id == PARENT_ENDPOINT then
+    return parent
   else
-    return "main"
+    return parent:get_child_by_parent_assigned_key(string.format("%02X", ep_id))
   end
 end
 
-local device_init = function(self, device)
-  device:set_component_to_endpoint_fn(component_to_endpoint)
-  device:set_endpoint_to_component_fn(endpoint_to_component)
+local function component_to_endpoint(device, component)
+  return PARENT_ENDPOINT
```

Comment:
Are we sure this function is necessary?  Does the device not join with a `fingerprinted_endpoint_id` of 1?

Reply:
Was a real device failing?  Or were tests failing?

Reply:
Yes, that was not needed, I deleted that mapping